lib/pdf_fill/extras_generator_v2.rb

```diff
@@ -5,6 +5,68 @@ class ExtrasGeneratorV2 < ExtrasGenerator
     HEADER_FONT_SIZE = 14.5
     SUBHEADER_FONT_SIZE = 10.5
 
+    class Question
+      attr_accessor :section_index, :overflow
+
+      def initialize(metadata)
+        @section_index = nil
+        @number = metadata[:question_num]
+        @subquestions = []
+        @overflow = false
+      end
+
+      def add_text(value, metadata)
+        @subquestions << { value:, metadata: }
+        @overflow ||= metadata.fetch(:overflow, true)
+      end
+
+      def sorted_subquestions
+        @subquestions.sort_by { |subq| [subq[:metadata][:question_suffix], subq[:metadata][:question_text]] }
+      end
+
+      def render(pdf, list_format: false)
+        sorted_subquestions.each do |subq|
+          value = subq[:value]
+          metadata = subq[:metadata]
+          prefix = "#{metadata[:question_num]}#{metadata[:question_suffix]}. #{metadata[:question_text].humanize}"
+          i = metadata[:i]
+          prefix += " Line #{i + 1}" if i.present?
+
+          if list_format
+            pdf.text("#{prefix}: <b>#{value}</b>", inline_format: true)
+          else
+            pdf.move_down(10)
+            pdf.text("#{prefix}:", { style: :normal })
+            pdf.text(value.to_s, { style: :bold })
+          end
+        end
+      end
+    end
+
+    class ListQuestion < Question
```

Comment:
A list-and-loop question is literally just a list (array) of questions, except they all share one question heading. And we can style them differently in an upcoming PR.